nailgun/entity_mixins.py

```diff
@@ -551,17 +552,30 @@ def to_json_dict(self):
         json_dct = {}
         for field_name, field in fields.items():
             if field_name in values:
-                if isinstance(field, OneToOneField):
-                    json_dct[field_name] = values[field_name].to_json_dict()
+                value = values[field_name]
+                if value is None:
+                    json_dct[field_name] = None
+                elif isinstance(field, OneToOneField):
+                    json_dct[field_name] = value.to_json_dict()
                 elif isinstance(field, OneToManyField):
                     json_dct[field_name] = [
-                        entity.to_json_dict() for entity in values[field_name]
+                        entity.to_json_dict() for entity in value
                     ]
                 else:
-                    json_dct[field_name] = to_json_serializable(
-                        values[field_name])
+                    json_dct[field_name] = to_json_serializable(value)
```

Comment:
I did the following test:

```python
In [1]: from nailgun.entity_mixins import to_json_serializable

In [2]: to_json_serializable(None)

In [3]: print(to_json_serializable(None))
None
```

There is no need to treat the value is None since the else clause will already take care of it.